### PR TITLE
Implement search feature

### PR DESCRIPTION
**Pull request by dbmina**

### CODE REVIEW

File: frontend/src/containers/Search.tsx [hunk 1]
```diff
@@ -1,7 +1,292 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import search from "../static/search.svg";
+import { getPostsAction, searchAction } from "../store/Post/postAction";
+import "../styles/components/Search.scss";
+import { SearchType } from "../store/Post/postInterfaces";
+import PostItem from "../components/PostItem";
+import { useSearchPostState } from "../hooks/usePostsState";
+//import { useSearchPostState } from "../hooks/usePostsState";
```
Comment (young924) [inline frontend/src/containers/Search.tsx:9]: Remove comments plz!

File: frontend/src/store/Post/postAction.ts [hunk 2]
```diff
@@ -42,6 +54,22 @@ export const cartPostAction = (postId: number, folderId: number) => {
   };
 };
 
+
+export const searchAction = (
+  searchForm: SearchForm,
+  callbackFunc: (value: boolean) => void
+) => {
+  console.log(searchForm)
```
Comment (young924) [inline frontend/src/store/Post/postAction.ts:62]: Remove console.log please!

File: frontend/src/store/Post/postAction.ts [hunk 3]
```diff
@@ -42,6 +54,22 @@ export const cartPostAction = (postId: number, folderId: number) => {
...
+  return (dispatch: Redux.Dispatch<PostDispatchType>) => {
+    console.log(searchForm);
```
Comment (young924) [inline frontend/src/store/Post/postAction.ts:64]: Same here :)

File: frontend/src/containers/Search.tsx [hunk 4]
```diff
@@ -1,7 +1,292 @@
...
+          season: userInputs.season,
+          location: userInputs.location,
+          days: userInputs.days,
+          theme: userInputs.theme,
+          transportation: userInputs.transportation,
+        },
+        (value) => SetSearch(value)
+      )
+    );
+    console.log(searched);
```
Comment (young924) [inline frontend/src/containers/Search.tsx:66]: Remove this please!

**Review by young924 — APPROVED** [review]

**Review by yeppin — APPROVED** [review]

**Pull request merged**